src/main/resources/liquibase/crop_changelog/14_4_0.xml

```diff
@@ -0,0 +1,100 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+				   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+				   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+		http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+	<changeSet author="clarysabel" id="v14.4.0-1">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="location" columnName="ldefault"/>
+			</not>
+		</preConditions>
+		<comment>Add ldefault column to location table</comment>
+		<addColumn tableName="location">
+			<column name="ldefault" type="TINYINT(1) UNSIGNED ZEROFILL" defaultValue="0" afterColumn="program_uuid">
+				<constraints nullable="false"/>
+			</column>
+		</addColumn>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v14.4.0-2">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<columnExists tableName="location" columnName="ldefault"/>
+				<sqlCheck expectedResult="0">
+					select count(1) from location l
+					where l.ldefault = 1 and l.ltype = (select u.fldno from udflds u where u.ftable='LOCATION' and u.ftype='LTYPE' and fcode='SSTORE')
+				</sqlCheck>
+			</and>
+		</preConditions>
+
+		<comment>Set DSS as default storage location</comment>
+
+        <sql dbms="mysql">
+			update location set ldefault = 1 where
+			ltype = (select u.fldno from udflds u where u.ftable='LOCATION' and u.ftype='LTYPE' and fcode='SSTORE') and labbr='DSS'
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v14.4.0-3">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<sqlCheck expectedResult="0">
+					select count(1) from ims_lot l
+					where l.locid is null;
+				</sqlCheck>
+			</not>
+		</preConditions>
+		<comment>Set the default storage location to the lots with no locations</comment>
+		<sql dbms="mysql">
+			update ims_lot set locid = (select l.locid from location l where l.ltype = (select u.fldno from udflds u where u.ftable='LOCATION' and u.ftype='LTYPE' and fcode='SSTORE') and l.ldefault=1)
+			where locid is null;
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v14.4.0-4">
+		<preConditions onFail="MARK_RAN">
```

Comment:
maybe this should be `WARN`? in case the _set default_ phase had any problems. Same with the others?

Reply:
will do

Reply:
@clarysabel provided some info:
- `WARN` -> not possible, changeset is executed anyway
- `CONTINUE` -> won't keep a record of the skipped changeset other than the missing row

Best option for now is to mark them as `MARK_RAN`